pkgs/top-level/all-packages.nix

```diff
@@ -3393,6 +3393,13 @@ in
     inherit (darwin) locale;
   };
 
+  gawk-with-extensions = callPackage ../tools/text/gawk/gawk-with-extensions.nix {
+    extensions = let
+      gawkextlib = callPackage ../tools/text/gawk/gawkextlib.nix {};
```

Comment:
This is a big step back. This prevents users from using extensions with setupHook like
```
nix-shell -p gawk -p gawk-extensions.json
```
And it also prevents users from doing `gawk-with-extensions.override { extensions = [ gawkextlib.json ] }` because you hid the orignal value of `gawkextlib`.

Reply:
ah, okay, reverting. I was trying to avoid polluting top-level